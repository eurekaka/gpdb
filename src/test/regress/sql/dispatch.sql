-- Misc tests related to dispatching queries to segments.

-- Test quoting of GUC values and databse names when they're sent to segments

-- There used to be a bug in the quoting when the search_path setting was sent
-- to the segment. It was not easily visible when search_path was set with a
-- SET command, only when the setting was sent as part of the startup packet.
-- Set search_path as a per-user setting so that we can test that.
CREATE DATABASE "dispatch test db";
ALTER DATABASE "dispatch test db" SET search_path="my schema",public;

\c "dispatch test db"

CREATE SCHEMA "my schema";

-- Create a table with the same name in both schemas, "my schema" and public.
CREATE TABLE "my table" (t text);
INSERT INTO "my table" VALUES ('myschema.mytable');

CREATE TABLE public."my table" (t text);
INSERT INTO public."my table" VALUES ('public.mytable');

SELECT t as unquoted FROM "my table";
SELECT t as myschema FROM "my schema"."my table";
SELECT t as public FROM public."my table";

DROP TABLE "my table";
DROP TABLE public."my table";

-- Create another table with the same name. To make sure the DROP worked
-- and dropped the correct table.
CREATE TABLE "my table" (id integer);
DROP TABLE "my table";

-- Clean up
\c regression
DROP DATABASE "dispatch test db";

--
-- test QD will report failure if QE fails to send its motion_listener back
-- during backend initialization
--

-- start_ignore
\! gpfaultinjector -f send_qe_details_init_backend -y reset -s 2
-- inject a 'skip' fault before QE sends its motion_listener
\! gpfaultinjector -f send_qe_details_init_backend -y skip -s 2 -o 0
-- end_ignore

-- terminate exiting QEs first
\c
-- verify failure will be reported
SELECT 1 FROM gp_dist_random('gp_id');

-- reset fault injector
-- start_ignore
\! gpfaultinjector -f send_qe_details_init_backend -y reset -s 2
-- end_ignore
