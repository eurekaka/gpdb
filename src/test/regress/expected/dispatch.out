-- Misc tests related to dispatching queries to segments.
-- Test quoting of GUC values and databse names when they're sent to segments
-- There used to be a bug in the quoting when the search_path setting was sent
-- to the segment. It was not easily visible when search_path was set with a
-- SET command, only when the setting was sent as part of the startup packet.
-- Set search_path as a per-user setting so that we can test that.
CREATE DATABASE "dispatch test db";
ALTER DATABASE "dispatch test db" SET search_path="my schema",public;
NOTICE:  schema "my schema" does not exist
\c "dispatch test db"
CREATE SCHEMA "my schema";
-- Create a table with the same name in both schemas, "my schema" and public.
CREATE TABLE "my table" (t text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 't' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO "my table" VALUES ('myschema.mytable');
CREATE TABLE public."my table" (t text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 't' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO public."my table" VALUES ('public.mytable');
SELECT t as unquoted FROM "my table";
     unquoted     
------------------
 myschema.mytable
(1 row)

SELECT t as myschema FROM "my schema"."my table";
     myschema     
------------------
 myschema.mytable
(1 row)

SELECT t as public FROM public."my table";
     public     
----------------
 public.mytable
(1 row)

DROP TABLE "my table";
DROP TABLE public."my table";
-- Create another table with the same name. To make sure the DROP worked
-- and dropped the correct table.
CREATE TABLE "my table" (id integer);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'id' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
DROP TABLE "my table";
-- Clean up
\c regression
DROP DATABASE "dispatch test db";
--
-- test QD will report failure if QE fails to send its motion_listener back
-- during backend initialization
--
-- start_ignore
\! gpfaultinjector -f send_qe_details_init_backend -y reset -s 2
20160823:15:12:59:015496 gpfaultinjector:localhost:gpadmin-[INFO]:-Starting gpfaultinjector with args: -f send_qe_details_init_backend -y reset -s 2
20160823:15:12:59:015496 gpfaultinjector:localhost:gpadmin-[INFO]:-local Greenplum Version: 'postgres (Greenplum Database) 4.3.99.00 build dev'
20160823:15:12:59:015496 gpfaultinjector:localhost:gpadmin-[INFO]:-Obtaining Segment details from master...
20160823:15:12:59:015496 gpfaultinjector:localhost:gpadmin-[INFO]:-Injecting fault on 1 segment(s)
20160823:15:12:59:015496 gpfaultinjector:localhost:gpadmin-[INFO]:-Injecting fault on localhost.localdomain:/home/gpadmin/workspace/data/single_debug/primary/gpseg0:content=0:dbid=2:mode=s:status=u
20160823:15:12:59:015496 gpfaultinjector:localhost:gpadmin-[INFO]:-DONE
-- inject a 'skip' fault before QE sends its motion_listener
\! gpfaultinjector -f send_qe_details_init_backend -y skip -s 2 -o 0
20160823:15:12:59:015508 gpfaultinjector:localhost:gpadmin-[INFO]:-Starting gpfaultinjector with args: -f send_qe_details_init_backend -y skip -s 2 -o 0
20160823:15:12:59:015508 gpfaultinjector:localhost:gpadmin-[INFO]:-local Greenplum Version: 'postgres (Greenplum Database) 4.3.99.00 build dev'
20160823:15:12:59:015508 gpfaultinjector:localhost:gpadmin-[INFO]:-Obtaining Segment details from master...
20160823:15:12:59:015508 gpfaultinjector:localhost:gpadmin-[INFO]:-Injecting fault on 1 segment(s)
20160823:15:12:59:015508 gpfaultinjector:localhost:gpadmin-[INFO]:-Injecting fault on localhost.localdomain:/home/gpadmin/workspace/data/single_debug/primary/gpseg0:content=0:dbid=2:mode=s:status=u
20160823:15:12:59:015508 gpfaultinjector:localhost:gpadmin-[INFO]:-DONE
-- end_ignore
-- terminate exiting QEs first
\c
-- verify failure will be reported
SELECT 1 FROM gp_dist_random('gp_id');
ERROR:  failed to acquire resources on one or more segments
DETAIL:  Internal error: No motion listener port (seg0 10.152.10.117:25432)
-- reset fault injector
-- start_ignore
\! gpfaultinjector -f send_qe_details_init_backend -y reset -s 2
20160823:15:12:59:015522 gpfaultinjector:localhost:gpadmin-[INFO]:-Starting gpfaultinjector with args: -f send_qe_details_init_backend -y reset -s 2
20160823:15:12:59:015522 gpfaultinjector:localhost:gpadmin-[INFO]:-local Greenplum Version: 'postgres (Greenplum Database) 4.3.99.00 build dev'
20160823:15:12:59:015522 gpfaultinjector:localhost:gpadmin-[INFO]:-Obtaining Segment details from master...
20160823:15:12:59:015522 gpfaultinjector:localhost:gpadmin-[INFO]:-Injecting fault on 1 segment(s)
20160823:15:12:59:015522 gpfaultinjector:localhost:gpadmin-[INFO]:-Injecting fault on localhost.localdomain:/home/gpadmin/workspace/data/single_debug/primary/gpseg0:content=0:dbid=2:mode=s:status=u
20160823:15:12:59:015522 gpfaultinjector:localhost:gpadmin-[INFO]:-DONE
-- end_ignore
